crud-subject

nuto17 opened the pull request.

nuto17 committed:
a lot of errors

nuto17 committed:
crud-subject and many-to-many relation

yegorowns commented on src/main/java/ru/nuto/studyproject/mapper/SubjectMapper.java:
```diff
@@ -0,0 +1,20 @@
+@Mapper(componentModel = "spring")
+public interface SubjectMapper {
+
+    @Mapping(target = "tasks", ignore = true)
```
аннотация лишняя, + расставь энтеры между методами

nuto17 replied:
done

yegorowns replied:
resolve

yegorowns commented on src/main/java/ru/nuto/studyproject/mapper/TaskMapper.java:
```diff
@@ -1,18 +1,53 @@
 import java.util.List;
 
 @Mapper(componentModel = "spring")
 public interface TaskMapper {
```
энтер после объявления тела маппера

nuto17 replied:
done

yegorowns replied:
resolve

yegorowns commented on src/main/java/ru/nuto/studyproject/service/SubjectService.java:
```diff
@@ -0,0 +1,34 @@
+
+@Service
+@RequiredArgsConstructor
+
```
тут убери лишний энтер

nuto17 replied:
done

yegorowns replied:
resolve

yegorowns commented on src/main/java/ru/nuto/studyproject/service/TaskService.java:
```diff
@@ -30,20 +31,23 @@ public void deleteTask(Long id) throws NotFoundTaskException {
+        return allTasks;
+    }
+
+
```
тут тож лишний энтер

nuto17 replied:
done

yegorowns replied:
resolve

nuto17 committed:
delete enters

nuto17 committed:
switch relation

nuto17 committed:
change logic

AlekcWins commented on src/main/java/ru/nuto/studyproject/controller/SubjectController.java:
```diff
@@ -0,0 +1,33 @@
+    public List<SubjectDto> getSubjectsDto(){
+        List<SubjectDto> subjectsDto = subjectService.getSubjects();
+        return subjectsDto;
+    }
```
захотел я удалить, захотел я изменить - что мне делать? 
crud не просто так называют

nuto17 replied:
done

AlekcWins replied:
resolve

AlekcWins commented on src/main/java/ru/nuto/studyproject/dto/TaskDto.java:
```diff
@@ -1,14 +1,15 @@
     private Long id;
 
-    private String subjectName;
+    private Long subjectId;
```
Использовать ДТО

nuto17 replied:
а зачем тут дто? мне же легче передавать id

AlekcWins replied:
затем что ты должен понимать кто будет использовать твои API - если говорить про фронтенд то че фронут вот пришел [1,2,3] - и че ему на это делать  3 запроса на получения каждой категории? - обычно такое делает либо на беке сразу, либо делают отдельный бекенд for frontend но общепринято делают что связанные сущности возвращаются в како то виде с инфой которая нужна  фронту

nuto17 replied:
я не понимаю как мне это сделать. я должен в создании таски сразу передавать все тело предмета? тогда зачем мне отдельная сущность объекта и две таблицы? но ты видимо имел в виду что-то другое, но что я не понял. про то что фронту будет непонятно что такое [n] это я согласен

nuto17 replied:
done

AlekcWins replied:
resolve

AlekcWins commented on src/main/java/ru/nuto/studyproject/model/Subject.java:
```diff
@@ -0,0 +1,29 @@
+
+    private String subjectName;
+
+    @OneToMany(mappedBy = "subject")
+    private List<Task> tasks;
```
про это говорили с тобой связь не здесь должна быть

nuto17 replied:
done

AlekcWins replied:
resolve

AlekcWins commented on src/main/java/ru/nuto/studyproject/service/SubjectService.java:
```diff
@@ -0,0 +1,33 @@
+        return subjectsDto;
+    }
+
+    public List<Subject> findSubjectsById(List<Long> ids){
+        List<Subject> subjects = subjectRepository.findAllById(ids);
+        return subjects;
+    }
```
очень старнный метод а зачем?

nuto17 committed:
change db logic, add crud for subject

AlekcWins commented on src/main/java/ru/nuto/studyproject/dto/SubjectDto.java:
```diff
@@ -0,0 +1,11 @@
+
+    private Long id;
+
+    private String subjectName;
```
просто name 
ибо это итак SubjectDto говорить к чему относится

nuto17 replied:
done

AlekcWins replied:
resolve

AlekcWins commented on src/main/java/ru/nuto/studyproject/model/Subject.java:
```diff
@@ -0,0 +1,23 @@
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    private String subjectName;
```
аналогично просто name

nuto17 replied:
done

AlekcWins replied:
resolve

nuto17 committed:
1.fixed dto logic
2.rename "subjectName" and "subjectNameDto"
3.add some logic in TaskService,SubjectService

AlekcWins commented on src/main/java/ru/nuto/studyproject/dto/TaskDto.java:
```diff
@@ -1,17 +1,17 @@
     private Long id;
 
-    private String subjectName;
+    private SubjectDto subjectDto;
```
проблема что название нужно subject

nuto17 replied:
можешь конкретнее, пожалуйста

AlekcWins replied:
private SubjectDto subject;

nuto17 replied:
done

AlekcWins replied:
resolve

nuto17 committed:
rename SybjectDto variable
